plugins/catalog-react/src/components/EntityPeekAheadPopover/EntityPeekAheadPopover.tsx

```diff
@@ -0,0 +1,237 @@
+      )}
+    </>
+  );
+};
+
+/**
+ * Shows an entity popover on hover of a component.
+ *
+ * @public
+ */
+const EntityCardActions = ({ entity }: { entity: Entity }) => {
+  const entityRoute = useRouteRef(entityRouteRef);
+
+  return (
+    <>
+      <Tooltip title="Show details">
+        <Link
+          component="button"
+          to={entityRoute({
+            name: entity.metadata.name,
+            namespace: entity.metadata.namespace || 'default',
+            kind: entity.kind.toLocaleLowerCase('en-US'),
+          })}
+        >
+          <InfoIcon color="action" />
+        </Link>
+      </Tooltip>
+    </>
+  );
+};
+
+/**
+ * Shows an entity popover on hover of a component.
+ *
+ * @public
+ */
+export const EntityPeekAheadPopover = ({
+  entityRef,
+  children,
+}: EntityPeekAheadPopoverProps) => {
+  const classes = useStyles();
+  const apiHolder = useApiHolder();
+  const popupState = usePopupState({
+    variant: 'popover',
+    popupId: 'entity-peek-ahead',
+  });
+
+  const [{ loading, error, value: entity }, load] = useAsyncFn(async () => {
+    const catalogApi = apiHolder.get(catalogApiRef);
+    if (catalogApi) {
+      const retrievedEntity = await catalogApi.getEntityByRef(entityRef);
+      if (!retrievedEntity) {
+        throw new Error(`${entityRef.name} was not found`);
+      }
+      return retrievedEntity;
+    }
+    return undefined;
+  }, [apiHolder, entityRef]);
+
+  useEffect(() => {
+    if (popupState.isOpen && !entity && !error && !loading) {
+      load();
+    }
+  }, [popupState.isOpen, load, entity, error, loading]);
+
+  return (
+    <>
+      <div data-testid="trigger" {...bindHover(popupState)}>
+        {children}
+      </div>
+      <HoverPopover
+        PaperProps={{
+          className: classes.popoverPaper,
+        }}
+        {...bindPopover(popupState)}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'center',
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'center',
+        }}
+      >
+        <Card>
+          {loading && <Progress />}
+          <CardContent>
+            <Typography color="textSecondary">{entityRef.namespace}</Typography>
+            <Typography variant="h5" component="div">
+              {entityRef.name}
+            </Typography>
+            {error && <Alert severity="warning">{error.message}</Alert>}
+            {entity ? (
+              <>
+                <Typography color="textSecondary">{entity.kind}</Typography>
+                <Typography className={classes.descriptionTypography} paragraph>
+                  {entity.metadata.description}
+                </Typography>
+                <Typography>{entity.spec?.type}</Typography>
+                <Box marginTop="0.5em">
+                  {(entity.metadata.tags || [])
+                    .slice(0, maxTagChips)
+                    .map(tag => {
+                      return <Chip size="small" label={tag} />;
```

Comment:
remember to give iterated things a `key` prop

Reply:
done.